platform/migration/platform-migration/src/main/java/org/codice/ddf/configuration/migration/ImportMigrationManagerImpl.java

```diff
@@ -0,0 +1,194 @@
+public class ImportMigrationManagerImpl implements Closeable {
+    private static final Logger LOGGER = LoggerFactory.getLogger(ImportMigrationManagerImpl.class);
+
+    private final MigrationReport report;
+
+    /**
+     * Holds import migration contexts for all migratables registered in the system keyed by the
+     * migratable id. <code>null</code> key is used to represent the system context.
+     */
+    private final Map<String, ImportMigrationContextImpl> contexts;
+
+    private final ZipFile zip;
+
+    private final String version;
+
+    private final String productVersion;
+
+    private final Path exportFile;
+
+    /**
+     * Creates a new migration manager for an import operation.
+     *
+     * @param report      the migration report where to record warnings and errors
+     * @param exportFile  the exported zip file
+     * @param migratables a stream of all migratables in the system
+     * @throws MigrationException       if a failure occurs while processing the zip file (the error
+     *                                  will not be recorded with the report)
+     * @throws IllegalArgumentException if <code>report</code> is <code>null</code> or if it is not
+     *                                  for an import migration operation or if <code>exportFile</code>
+     *                                  or <code>migratables</code> is <code>null</code>
+     */
+    public ImportMigrationManagerImpl(MigrationReport report, Path exportFile,
+            Stream<? extends Migratable> migratables) {
+        this(report, exportFile, migratables, ImportMigrationManagerImpl.newZipFileFor(exportFile));
+    }
+
+    ImportMigrationManagerImpl(MigrationReport report, Path exportFile,
+            Stream<? extends Migratable> migratables, ZipFile zip) {
+        Validate.notNull(report, "invalid null report");
+        Validate.isTrue(report.getOperation() == MigrationOperation.IMPORT,
+                "invalid migration operation");
+        Validate.notNull(migratables, "invalid null migratables");
+        this.report = report;
+        this.exportFile = exportFile;
+        this.zip = zip;
+        Map<String, Object> metadata;
+
+        try {
+            // pre-create contexts for all registered migratables
+            this.contexts = migratables.collect(Collectors.toMap(Migratable::getId,
+                    m -> new ImportMigrationContextImpl(report, zip, m),
+                    ConfigurationMigrationManager.throwingMerger(),
+                    LinkedHashMap::new)); // to preserved ranking order and remove duplicates
+            // add a system contexts
+            contexts.put(null, new ImportMigrationContextImpl(report, zip));
+            zip.stream()
+                    .map(ze -> new ImportMigrationEntryImpl(this::getContextFor, ze))
+                    .forEach(me -> me.getContext()
+                            .addEntry(me));
+            metadata = retrieveMetadata();
+        } catch (IOException e) {
```

Comment:
:question: do we need to close the zip if we get an IOException?

Reply:
Good catch